apps/api-extractor/src/generators/DeclarationReferenceGenerator.ts

```diff
@@ -217,10 +200,12 @@ export class DeclarationReferenceGenerator {
       followedSymbol = this._collector.typeChecker.getExportSymbolOfSymbol(followedSymbol);
     }
     if (followedSymbol.flags & ts.SymbolFlags.Alias) {
-      const nextFollowedSymbol: ts.Symbol = this._collector.typeChecker.getAliasedSymbol(followedSymbol);
-      // do not follow alias to namespace, as it results in a source file that no longer knows its short name:
-      if (!(nextFollowedSymbol.flags & ts.SymbolFlags.Namespace)) {
-        followedSymbol = nextFollowedSymbol;
+      followedSymbol = this._collector.typeChecker.getAliasedSymbol(followedSymbol);
+
+      // Without this logic, we end up following the symbol `ns` in `import * as ns from './file'` to
+      // the actual file `file.ts`. We don't want to do this, so revert to the original symbol.
+      if (followedSymbol.flags & ts.SymbolFlags.ValueModule) {
+        followedSymbol = symbol;
       }
```

Comment:
Originally I couldn't figure out why this was needed, but after debugging, I understood what your original comment meant. I'm not sure if the logic you/I added is always what we want... but it seems to work well enough.

Reply:
It was hard to explain. A namespace that is created by a *-import is resolved to the imported module, which of course does not know the name under which it is exported.